routes/users.js

```diff
@@ -119,4 +130,53 @@ router.get("/list/:page", async (req, res, next) => {
   }
 });
 
+router.post("/address", validationCheck, async (req, res, next) => {
+  const { address, city, state, pinCode, phoneNumber } = req.body;
+  try {
+    const user = await Address.findOne({ user_id: req.user_id });
+    if (user)
+      return res.status(400).json({
+        success: 0,
+        message: "Your record is already present at our end",
+        data: user,
+      });
+    const userAddress = new Address({
+      user_id: req.user_id,
+      address: address.split(","),
+      city,
+      state,
+      pinCode,
+      phoneNumber,
+    });
+    await userAddress.save();
+    res.status(201).json({
+      success: 1,
+      message: "Address field has been created successfully",
+      data: userAddress,
+    });
+  } catch (err) {
+    next(err);
+  }
+});
+
+router.get("/get/:userid", async (req, res, next) => {
+  try {
+    const userData = await Address.findOne({
+      user_id: req.params.userid,
+    }).populate("user_id");
```

Comment:
populate `address` with `user`  and you are doing reverse, you are populating `user`